packages/message/src/Providers.ts

```diff
@@ -1,18 +1,37 @@
 import { Message } from '@cumulus/types';
-
+import { isCumulusMessageLike } from './CumulusMessage';
 type MessageProvider = {
   id: string
   protocol: string
   host: string
   port?: number
 };
 
+export const isMessageProvider = (
+  obj: any
+): obj is MessageProvider => (
+  obj instanceof Object
+  // in testing instanceof String can return false for string literals
+  && 'id' in obj && (typeof (obj.id) === 'string' || obj.id instanceof String)
+  && 'protocol' in obj && (typeof (obj.protocol) === 'string' || obj.protocol instanceof String)
+  && 'host' in obj && (typeof (obj.host) === 'string' || obj.host instanceof String)
+  && ('port' in obj ? (typeof (obj.port) === 'number' || obj.port instanceof Number) : true)
+);
+
 type MessageWithProvider = Message.CumulusMessage & {
   meta: {
     provider?: MessageProvider
   }
 };
 
+export const isMessageWithProvider = (
+  obj: any
+): obj is MessageWithProvider => (
+  isCumulusMessageLike(obj)
+  && 'meta' in obj
```

Comment:
sort of:
that syntax generally doesn't satisfy typescript. in *this* case it will becuase 'isCumulusMessageLike' has the effect of stating that obj.meta.provider exists *buuuut* it doesn't actually check because it's doing that bare check. so while it'll *let* me get away with it, i've kept with the pattern as if it wont

Reply:
`in this case it will becuase 'isCumulusMessageLike' has the effect of stating that obj.meta.provider exists` -- I'm not sure I follow your meaning, given isCumulusMessageLike only validates `cumulus_meta` is in the object but doesn't have any impact on the typing at all other than the boolean result.    

isMessageProvider is a guard on if it's `MessageProvider`, which should have the correct boolean for `obj?.meta` when `meta` isn't defined (which is `isMessageWithProvider(undefined)` or `false`), and the same for `obj?.meta?.provider` when it's not defined, which is stating all three assertions and eliminating the extra 'is it an object check' on obj.meta (if obj.meta isn't, it passes in the literal and fails `obj instanceof Object`). 

Apologies if I'm being obtuse.

Reply:
I don't think so, I'll change it